langchain-core/src/language_models/chat_models.ts

```diff
@@ -293,6 +296,104 @@ export abstract class BaseChatModel<
     return output;
   }
 
+  /** @ignore */
+  async _generateCached(
+    messages: BaseMessageLike[][],
+    cache: BaseCache<Generation[]>,
+    parsedOptions: this["ParsedCallOptions"],
+    handledOptions: RunnableConfig
```

Comment:
Wouldn't mind just making this an object parameter with instead of multiple args